### PR TITLE
chore: Convert singleton reconcilers to operatorpkg

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes #N/A 

**Description**

This updates our singleton controllers to use the singleton constructs in the `operatorpkg` package. This enables us to drop the implementation of the singleton builder that we currently have in Karpenter

**How was this change tested?**

`make presubmit`
`/karpenter snapshot`

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

File: test/suites/integration/kubelet_config_test.go [hunk 1]
```diff
@@ -157,7 +157,7 @@ var _ = Describe("KubeletConfiguration Overrides", func() {
 		// Get the DS pod count and use it to calculate the DS pod overhead
 		dsCount := env.GetDaemonSetCount(nodePool)
 		nodePool.Spec.Template.Spec.Kubelet = &corev1beta1.KubeletConfiguration{
-			MaxPods: lo.ToPtr(int32(1 + int32(dsCount))),
+			MaxPods: lo.ToPtr(1 + int32(dsCount)),
```
Comment (jmdeal) [inline test/suites/integration/kubelet_config_test.go:160]: Is this just a change for the linter?

Comment (jonathan-innis) [reply test/suites/integration/kubelet_config_test.go:160]: Yep

Comment (jonathan-innis) [reply test/suites/integration/kubelet_config_test.go:160]: It's the same either way

**Review by jmdeal — APPROVED** [review]
LGTM 🚀

**Pull request merged**